hackIDE/static/hackIDE/js/custom.js

```diff
@@ -751,4 +751,282 @@ $(document).ready(function(){
 	}
 
 
+	$("#register").click(function(){
+		var username = $('#signup_username').val();
+		var email = $("#signup_email").val();
+		var password = $("#signup_password").val();
+
+		$("#register").html("Registering");
+		$("#register").prop('disabled', true);
+		
+		var csrf_token = $(":input[name='csrfmiddlewaretoken']").val();
+		var form_data = {username : username, email : email, password : password, csrfmiddlewaretoken : csrf_token};
+		$.ajax({
+			url : 'register/',
+			type : 'POST',
+			data : form_data,
+			dataType : 'json',
+			timeout : '10000',
+			success : function(response){
+				$("#error_username").html(response.error_username);
+				$("#error_email").html(response.error_email);
+				$("#error_password").html(response.error_password);
+
+				$("#register").prop('disabled', false);
+				$("#register").html("Register");
+
+				$("#msg").html(response.msg);
+				if (response.error_username == "" && response.error_email == "" && response.error_password == "")
+				{
+					$("#register").html("Registered");
+					location.reload();
+				}
+			},
+		});
+	});
+
+	$('#login').click(function(){
+		var username = $('#login_username').val();
+		var password = $("#login_password").val();
+
+		$("#login").html("Logging In");
+		$("#login").prop("disabled", true);
+		var csrf_token = $(":input[name='csrfmiddlewaretoken']").val();
+
+		var login_form_data = {username : username, password : password, csrfmiddlewaretoken : csrf_token};
+
+		$.ajax({
+			url : 'login/',
+			type : 'POST',
+			data : login_form_data,
+			dataType : 'json',
+			timeout : 10000,
+			success : function(response){
+
+				$("#login").prop("disabled", false);
+				$("#login").html("Login");
+
+				$('#login_msg').html(response.msg);
+				if(response.msg != "Invalid credentials")
+				{
+					$("#login").html("Logged In Sucessfully");
+					location.reload();
+				}
+
+			},
+		});
+	});
+
+	$("#logout").click(function(){
+		var csrf_token = $(":input[name='csrfmiddlewaretoken']").val();
+		logout_data = {csrfmiddlewaretoken:csrf_token}		
+
+		$.ajax({
+			url : 'logout/',
+			type : 'POST',
+			data : logout_data,
+			dataType : 'json',
+			timeout : 10000,
+			success : function(response){
+				location.reload();
+			},
+		});
+	});
+
+	$("#title-save").click(function(){
+
+		// if a run request is ongoing
```

Comment:
Since all the code in this callback function is repeated, can you put it into a separate function and call it everywhere this code is repeated?